Add "Browser level lazy loading for CMSs" (#4232)
GoogleChrome/web.dev: src/site/content/en/blog/browser-level-lazy-loading-for-cmss/index.md

---
layout: post
title: Browser-level lazy-loading for CMSs
subhead: Learnings for adopting the standardized loading attribute
authors:
  - felixarntz
date: 2020-11-16
#updated: 2020-11-16

> hero: hero.jpg

Reviewer: Please resize to 3200 px width and optimize with tinyjpg.com

Author: Done!

alt: A lazy leopard relaxing on a tree
description: |
  This post provides guidance on how to adopt the loading
  attribute in content management systems.
tags:
  - blog # blog is a required tag for the article to show up in the blog.
  - performance
---

My goal with this post is to persuade CMS platform developers and contributors
(i.e. the people who develop CMS cores) that

> [now is the time to implement support for the browser-level lazy-loading feature](#the-case-for-implementing-lazy-loading-now).

Reviewer: ```suggestion
[now is the time to implement support for the browser-level image lazy-loading feature](#the-case-for-implementing-lazy-loading-now).
```

[power about 60% of websites](https://w3techs.com/technologies/overview/content_management),
so these platforms play a vital role in bringing adoption of modern browser
features to the web. With a few popular open-source CMSs such as
[WordPress](https://make.wordpress.org/core/2020/07/14/lazy-loading-images-in-5-5/),
[Joomla](https://github.com/joomla/joomla-cms/pull/28838), and
[TYPO3](https://review.typo3.org/c/Packages/TYPO3.CMS/+/63317) having already
implemented support for the `loading` attribute on images, let's have a look at
their approaches and the takeaways which are relevant for adopting the feature
in other CMS platforms as well. Lazy-loading media is a key web performance
feature that sites should benefit from at a large scale, which is why adopting
it at the CMS core level is recommended.

## The case for implementing lazy-loading now

### Standardization

Adoption of non-standardized browser features in CMSs facilitates widespread
testing and can surface potential areas of improvement. However, the general
consensus across CMSs is that, as long as a browser feature is not standardized,
it should preferably be implemented in the form of an extension or plugin for

> the respective platform. Only once standardized a feature can be considered for

Reviewer: ```suggestion
the respective platform. Only once standardized can a feature be considered for
```

and
[`iframe`](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#attr-iframe-loading)
elements.
{% endAside %}

### Browser support

Browser support of the feature is a similar concern: The majority of CMS users
should be able to benefit from the feature. If there is a considerable
percentage of browsers where the feature is not yet supported, the feature has
to ensure that it at least has no adverse effect for those.

{% Aside 'success' %}
[Browser-level lazy-loading is widely supported by browsers](https://caniuse.com/#feat=loading-lazy-attr)
and
[the `loading` attribute is simply ignored](/browser-level-image-lazy-loading/#browser-compatibility)
by those browsers that have not adopted it yet.
{% endAside %}

### Distance-from-viewport thresholds

A common concern with lazy-loading implementations is that they in principle
increase the likelihood that an image will not be loaded once it becomes visible
in the user's viewport because the loading cycle starts at a later stage.
Contrary to previous JavaScript-based solutions,
[browsers approach this conservatively](/browser-level-image-lazy-loading/#distance-from-viewport-thresholds)

> and furthermore can fine-tune their approach based on real-world heuristics in
> the field, minimizing the impact, so browser-level lazy-loading should be safe to adopt

Reviewer: ```suggestion
and furthermore can fine-tune their approach based on real-world user experience data,
minimizing the impact, so browser-level lazy-loading should be safe to adopt
```
At first I came here to edit "real-world heuristics in the field" for redundancy (real-world and in the field mean the same thing) but upon looking at this a bit further I think we mean data, not heuristics. Because aren't heuristics rule-of-thumb guidelines that you use to make decisions? Doesn't quite seem like the right word here.

Author: Makes sense, "data" is a more accurate term here.
